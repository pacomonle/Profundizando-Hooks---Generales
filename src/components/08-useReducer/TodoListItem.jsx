import React from 'react'
import './styles.css'


const TodoListItem = ({todo, handleDelete, handleToggle, index}) => {
    return (
        <li 
        className='list-group-item'
        >
            <h4  
            className={todo.done && 'complete'}
            onClick = {() => handleToggle(todo.id)}>
            {index + 1}. {todo.desc} 
            </h4>  
            <button 
            type="button"
            onClick={() => handleDelete(todo.id)} 
            className='btn btn-danger'>
            Borrar
            </button>
        </li>
    )
}

export default TodoListItem
