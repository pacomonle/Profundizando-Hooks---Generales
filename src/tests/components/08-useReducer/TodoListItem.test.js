import React from 'react';
import { shallow } from 'enzyme';
import TodoListItem  from '../../../components/08-useReducer/TodoListItem';
import { demoTodos } from '../../fixtures/demoTodos';
import '@testing-library/jest-dom'

//demoTodos

describe('Pruebas en <TodoListItem />', () => {

 // simulacion de las function de las props
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    
    // simulacion de las props
    const index= 0
    const wrapper = shallow(
        <TodoListItem 
            todo={ demoTodos[0] }
            index={ index}
            handleDelete={ handleDelete }
            handleToggle={ handleToggle }
        />
    )



    test('debe de mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de llamar la función handleDelete', () => {
        
        wrapper.find('button').simulate('click');
        expect( handleDelete ).toHaveBeenCalledWith( demoTodos[0].id );

    });

    test('debe de llamar la función handleToggle', () => {
        
        wrapper.find('h4').simulate('click');
        expect( handleToggle ).toHaveBeenCalledWith( demoTodos[0].id );

    });

    test('debe de mostrar el texto correctamente', () => {
        
        const h4 = wrapper.find('h4');
       //  console.log(index)
        expect( h4.text().trim() ).toBe(`${index + 1}. ${ demoTodos[0].desc }`)

    });

    test('debe de tener la clase complete si el TODO.done = true', () => {
        // simulamos el cambio en el primer todo
        const todo = demoTodos[0];
        todo.done = true;
        
        const wrapper = shallow(
            <TodoListItem 
                todo={ todo }
            />
        );
        console.log(wrapper.html())
        expect(wrapper.find('h4').hasClass('complete') ).toBe(true);


    });
    
    
})
