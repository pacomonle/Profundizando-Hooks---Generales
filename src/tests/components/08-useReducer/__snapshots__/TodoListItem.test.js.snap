// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item"
>
  <h4
    className={false}
    onClick={[Function]}
  >
    1
    . 
    Aprender React
  </h4>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    type="button"
  >
    Borrar
  </button>
</li>
`;
